src/app/locator/qgsgotolocatorfilter.cpp

```diff
@@ -87,51 +88,114 @@ void QgsGotoLocatorFilter::fetchResults( const QString &string, const QgsLocator
     }
   }
 
-  if ( okX && okY )
+  if ( okX && okY )  // Ok found a valid looking x,y coordinate pair, check against Wgs84crs, MapCanvas-crs AND layer-crs's
   {
-    QVariantMap data;
-    const QgsPointXY point( posX, posY );
-    data.insert( QStringLiteral( "point" ), point );
-
-    const bool withinWgs84 = wgs84Crs.bounds().contains( point );
-    if ( !posIsDms && currentCrs != wgs84Crs )
+    // collect all crs's from all layers (plus MapCanvas crs and wgs84)
+    QList<QString> crsList;  // TODO make this a list
+    crsList.append( wgs84Crs.authid() );
```

Comment:
we should avoid storing QgsCoordinateReferenceSystem objects as strings -- just store them in a QList< QgsCoordinateReferenceSystem > instead.

Reply:
done in next commit

Reply:
You could also use a QSet instead of a QList and skip checking if CRSs should be added or not